ci/steps.py

```diff
@@ -97,11 +100,17 @@ def task_step_script(
         args = ','.join((
             f"{param.name.replace('-', '_')}='$(params.{param.name})'" for param in params
         ))
+        if results:
+            args += ',' + ','.join((
+                f"{result.name.replace('-', '_')}='$(results.{result.name}.path)'" for result in results
+            ))
```

Comment:
Consider something similar to
```suggestion
            result_mapping_strings = [
                f"{result.name.replace('-', '_')}='$(results.{result.name}.path)'"
                for result in results
            ]
            args = ','.join([args, ','.join(result_mapping_strings)])
```
though that admittedly doesn't improve it much. I'll leave it up to you.

Reply:
might be a bit more readable